Add -? option to Help in dotnet test (#45141)
dotnet/docs: docs/core/tools/dotnet-test.md

  The short form `-p` can be used for `--property`. The same applies for `/property:property=value` and its short form is `/p`.
  More informatiom about the available arguments can be found in [the dotnet msbuild documentation](dotnet-msbuild.md).

- **`-?|-h|--help`**

>   Prints out a description of how to use the command. There are static options along with platform and extension options. Extension options are dynamic and might differ from one test application to another, as they are based on the registered extensions in the test project.

Reviewer: ```suggestion
  Prints out a description of how to use the command. Some options are static while others, namely platform and extension options, are dynamic. These dynamic options might differ from one test application to another, as they are based on the registered extensions in the test project.
```
